src/main/java/com/libraryproject/librarysystem/controller/BookController.java

```diff
@@ -0,0 +1,32 @@
+package com.libraryproject.librarysystem.controller;
+
+import com.libraryproject.librarysystem.dto.BookResponseDto;
+import com.libraryproject.librarysystem.service.BookService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/books")
+public class BookController {
```

Comment:
oba endpointy są poprawne. Tak jak pisałaś, pagination endpoint jest używany aby nie zalać np UI milionem danych (ponad to taki endpoint długo się ładuje). Paginacja oszczędza czas i ułatwia pracę.

Ponieważ nie mamy aplikacji konsumującej nasz API taki performance nie jest aż tak istotny, jednakże jest to fajne inicjatywa z Twojej strony! ja bym zostawiła oba endpointy i dalej tak rozwijała aplikację (dając możliwość wyboru). Drobna uwaga - paginacja zazwyczaj daje też możliwość wyboru rozmiaru strony, u Ciebie w tej chwili jest to sztywno ustalone 10 "jednostek". Możesz to zmienić jeśli masz ochotę, przyjmując dodatkowy ReqestParam

Reply:
W takim razie zostawiam 2 endpointy + zmieniłam pageSize z sztywnej zmiennej na RequestParam